supernode/daemon/mgr/progress/superload_manager.go

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright The Dragonfly Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package progress
+
+import (
+	"context"
+	"time"
+
+	"github.com/dragonflyoss/Dragonfly/pkg/errortypes"
+)
+
+const (
+	// renewInterval is the interval time to check the superload.
+	renewInterval = 2 * time.Second
+
+	// renewDelayTime if the superload has not been changed after renewDelayTime, it should be renewed.
+	renewDelayTime = 30 * time.Second
+)
+
+// UpdateSuperLoad updates the superload of taskID by adding the delta.
+// The updated will be `false` if failed to do update operation.
+//
+// It's considered as a failure when then superload is greater than limit after adding delta.
+func (pm *Manager) UpdateSuperLoad(ctx context.Context, taskID string, delta, limit int32) (updated bool, err error) {
+	loadState, err := pm.superLoad.getAsSuperLoadState(taskID)
+	if err != nil {
+		if !errortypes.IsDataNotFound(err) {
+			return false, err
+		}
+
+		// init the value for taskID when not found.
+		pm.superLoad.add(taskID, newSuperLoadState())
+	}
+
+	if loadState.loadValue.Add(delta) > limit && limit > 0 {
```

Comment:
The `loadState` maybe `nil`. It's not the object we created above when it's not exist before. We should get it again from map after adding. Otherwise we may use different objects with same `taskId` in multi-goroutines. It's better to make the `get` and `add` as one atomic operation.

Reply:
Done.